Random failuress in teardown of TestCliWebServer.test_cli_webserver_foreground

We have some random failures in teardown of TestCliWebServer.test_cli_webserver_foreground.

Since this is in teardown, it can be likely ignored because we are killing those background processes with -9 anyway.

https://github.com/apache/airflow/pull/11526/checks?check_run_id=1268530821#step:6:1193

```
________________ TestCliWebServer.test_cli_webserver_foreground ________________

self = <tests.cli.commands.test_webserver_command.TestCliWebServer testMethod=test_cli_webserver_foreground>

    def tearDown(self) -> None:
>       self._check_processes()

tests/cli/commands/test_webserver_command.py:247: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.cli.commands.test_webserver_command.TestCliWebServer testMethod=test_cli_webserver_foreground>

    def _check_processes(self):
        # Confirm that webserver hasn't been launched.
        # pgrep returns exit status 1 if no process matched.
        exit_code_pgrep_webserver = subprocess.Popen(["pgrep", "-c", "-f", "airflow webserver"]).wait()
        exit_code_pgrep_gunicorn = subprocess.Popen(["pgrep", "-c", "-f", "gunicorn"]).wait()
        if exit_code_pgrep_webserver != 1 or exit_code_pgrep_gunicorn != 1:
            subprocess.Popen(["ps", "-ax"]).wait()
            if exit_code_pgrep_webserver != 1:
                subprocess.Popen(["pkill", "-9", "-f", "airflow webserver"]).wait()
            if exit_code_pgrep_gunicorn != 1:
                subprocess.Popen(["pkill", "-9", "-f", "gunicorn"]).wait()
    
            raise AssertionError(
>               "Background processes are running that prevent the test from passing successfully."
            )
E           AssertionError: Background processes are running that prevent the test from passing successfully.

tests/cli/commands/test_webserver_command.py:243: AssertionError
```

--- a/tests/cli/commands/test_webserver_command.py
+++ b/tests/cli/commands/test_webserver_command.py
@@ -227,7 +227,7 @@ class TestCliWebServer(unittest.TestCase):
         self._check_processes()
         self._clean_pidfiles()
 
-    def _check_processes(self):
+    def _check_processes(self, ignore_running=False):
         # Confirm that webserver hasn't been launched.
         # pgrep returns exit status 1 if no process matched.
         exit_code_pgrep_webserver = subprocess.Popen(["pgrep", "-c", "-f", "airflow webserver"]).wait()
@@ -238,13 +238,13 @@ class TestCliWebServer(unittest.TestCase):
                 subprocess.Popen(["pkill", "-9", "-f", "airflow webserver"]).wait()
             if exit_code_pgrep_gunicorn != 1:
                 subprocess.Popen(["pkill", "-9", "-f", "gunicorn"]).wait()
-
-            raise AssertionError(
-                "Background processes are running that prevent the test from passing successfully."
-            )
+            if not ignore_running:
+                raise AssertionError(
+                    "Background processes are running that prevent the test from passing successfully."
+                )
 
     def tearDown(self) -> None:
-        self._check_processes()
+        self._check_processes(ignore_running=True)
         self._clean_pidfiles()
 
     def _clean_pidfiles(self):

